### DISCUSSION TITLE
[Test-DbaConnection] Logon failure: the user has not been granted the requested logon type at this computer

### DISCUSSION THREAD

**Discussion by nojnoj33** [dataplat/dbatools#8819] [General] [answered]
Hi,

I am trying to use DBAtools plugin with a service account, so I can connect to multiple Prod servers and limit the level of SQL permissions.
I keep getting the below error when using the service account and -SqlCredential 
The Error =  [Test-DbaConnection] Failure | Logon failure: the user has not been granted the requested logon type at this computer 

The server account has 'Log on as a service' & 'Allow log on locally' local policy rights.

Testing.......... What am i missing?
Using my own account works fine, I have tested adding the service account as a SA in SQL. but this seems line a local rights issue.
The service account is in a sub group in the Local Administrators group on the test server
$winCred = my account
$Credentials = service account.
This Fails - Test-DbaConnection -SqlInstance <Server> -SqlCredential $Credentials -Credential $winCred
This works:  Test-DbaConnection -SqlInstance <Server> -SqlCredential $winCred -Credential $Credentials 

[Test-DbaConnection] Failure | Logon failure: the user has not been granted the requested logon type at this computer

Any ideas please.

**Comment by wsmelton** [comment]
Have you tried searching for that error text: `the user has not been granted the requested logon type at this computer`?

It pertains to your policy settings of the server(s).

**Reply by nojnoj33** [reply]
Well Yes!. 
its suggested to add the server account to have 'Log on as a service' & 'Allow log on locally' local policy rights.
I have done this. If I set my account (SQL SA & Server local Administrator) -SqlCredential works ok.

**Reply by wsmelton** [reply]
A few other suggestions: 
- [Ref](https://learn.microsoft.com/en-us/answers/questions/760241/the-user-has-not-been-granted-the-requested-logon)
- [Ref](https://techdirectarchive.com/2022/02/08/error-1385-the-user-has-not-been-granted-the-requested-logon-type-at-this-time/)

Verify the `Access this computer from the network` policy is not blocking the account.

**Comment by potatoqualitee** [comment, marked as answer]
I've had a similar issue on a DISA STIG'd network that is due to a network-related GPO.  I believe, like Shawn said, that it's `deny access to this computer from the network`

https://www.stigviewer.com/stig/windows_10/2016-11-03/finding/V-63871
